### PR TITLE
Add Modpack-page to instances that were created from files

### LINKED ISSUE #1394: Allow updating instances through mrpack
### Role

I make a modpack I want to update locally

### Suggestion

I want an updater for local mrpacks, instead of requiring modrinth publishing

### Benefit

So it's not required to publish the mrpack to modrinth

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

Okay, imagine this:

You import an instance through an mrpack

You want to update it, so you use another, separate mrpack, which is a more up-to-date/newer version of the first one

Prism Launcher allows you to update the existing instance through the new mrpack, copying over the options.txt, config, any custom-added mods, resourcepacks, preferences, whatever is done with the current modrinth modpack updater.

You don't need to publish it to modrinth, so you can use it to test beta versions/run long-running mutations of your modpack.

### LINKED ISSUE #1731: Add a hyperlink to the modpack page in the CurseForge tab of the console window
### Role

I enjoy downloading modpacks and playing them occasionally.

### Suggestion

I hope there is a hyperlink to the modpack page in the CurseForge tab of the console window.

### Benefit

so that I can visit their page without manually googling them.

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

Actually, Modrinth modpacks already have a hyperlink to their page on the Modrinth tab.

### LINKED ISSUE #1813: Canceling the update modpack from file creates a bugged popup
### Operating System

Windows

### Version of Prism Launcher

Prism Launcher 8.0

### Version of Qt

Qt 6.6.0

### Description of bug

Cancelling the file selection when updating a mod pack from file creates a bugged progress popup.

### Steps to reproduce

- Add a new instance from a modpack, such as Fabulously Optimized.
- Edit the modpack and go to the CurseForge / Modrinth tab. 
- Click "Update from file" on the right side.
- On the file selection, click Cancel or close the window.

### Suspected cause

_No response_

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

### PR DESCRIPTION
**Pull request by Trial97**
Parrent PR: https://github.com/PrismLauncher/PrismLauncher/pull/1405
should fix: https://github.com/PrismLauncher/PrismLauncher/issues/1394#issuecomment-1767449001
From that comment looks like I did not fix the original issue just extended the functionality of the managed packs(downloaded through prism from modrinth/curseforge)
This PR aims to have a page for local packs (imported from files).
This only works with CurseForge and Modrinth packs.

How to test:

    download a modrinth pack( an older version)
    install it using the mrpack file
    download a newer version of the pack(outside of prism)
    click edit instance on the created instance
    you should have a similar page as when that pack would be downloaded through prism
    entering that page you will be greeted with:
        the pack name(the name of the instance)
        a big button with upload from file
        a changelog that explains this is a local pack and can only be updated through respective format
    click on the upload from file
    select the new version
    profit

Also fixes #1731
fixes #1813

### CODE REVIEW

**Comment by Trial97** [comment]
This PR and https://github.com/PrismLauncher/PrismLauncher/pull/1727 are self-exclusive once one is accepted the other one should be closed.

**Review by DioEgizio — APPROVED** [review]
i personally prefer this approach over the other one but idk what the other mantainers think

**Comment by Trial97** [comment]
Regarding the ui: the button is gigantic but I have zero talent for UX can you suggest an alternative(a mockup)
Also regarding on a way to link to a modrinth/curseforge pack a solution can be found in: hhttps://github.com/PrismLauncher/PrismLauncher/pull/1414 where once you import a file we try to detect if is a curseforge/modrinth pack and if yes just make it one(right now works only for modrinth)

File: launcher/ui/pages/instance/ManagedPackPage.cpp [hunk 1]
```diff
@@ -471,8 +489,7 @@ void FlameManagedPackPage::parseManagedPack()
 
 QString FlameManagedPackPage::url() const
 {
-    // FIXME: We should display the websiteUrl field, but this requires doing the API request first :(
-    return {};
+    return "https://www.curseforge.com/projects/" + m_inst->getManagedPackID();
```
Comment (Scrumplex) [inline launcher/ui/pages/instance/ManagedPackPage.cpp:493]: Perhaps leave the FIXME here, as CurseForge could change their URL structure in the future.

**Review by Scrumplex — CHANGES_REQUESTED** [review]

File: launcher/ui/pages/instance/ManagedPackPage.cpp (hunk 1 above)
Comment (Trial97) [reply launcher/ui/pages/instance/ManagedPackPage.cpp:493]: unlikely to happen, but just to be sure I added it back

**Review by Scrumplex — APPROVED** [review]

**Pull request merged**